activerecord/lib/active_record/query_logs.rb

```diff
@@ -151,7 +151,9 @@ def tag_content
             end
 
             self.formatter.format(key, val) unless val.nil?
-          end.join(",")
+          end
+          pairs.sort! if self.formatter.sort_pairs
```

Comment:
```suggestion
          pairs.sort_by!(&:first) if self.formatter.sort_pairs
```

Reply:
Ah wait, `pairs` is `Array[String]` here? Might be better to sort by keys before formatting them.

Reply:
Following the SQLCommenter spec down to the letter, sorting happens after serializing `key=value` pairs. Sorting an array of strings is also faster on my M1 Pro w/ Ruby 3.1.2:

```ruby
require 'benchmark/ips'

HASH = [
    traceparent: '00-5bd66ef5095369c7b0d1f8f4bd33716a-c532cb4098ac3dd2-01',
    tracestate: 'congo%3Dt61rcWkgMzE%2Crojo%3D00f067aa0ba902b7',
    route: '%2Fparam*d',
    controller: 'index',
    application: 'Rails',
    foo: 'bar',
]
ARRAY = HASH.map { |k, v| "#{k}='#{v}'"}

Benchmark.ips do |x|
  x.report("sort hash") { HASH.sort_by!(&:first) }
  x.report("sort string") { ARRAY.sort! }
  x.compare!
end
```

```
Warming up --------------------------------------
           sort hash   165.692k i/100ms
         sort string     1.752M i/100ms
Calculating -------------------------------------
           sort hash      1.665M (± 0.8%) i/s -      8.450M in   5.075960s
         sort string     17.530M (± 0.3%) i/s -     89.332M in   5.096062s

Comparison:
         sort string: 17529797.2 i/s
           sort hash:  1664883.4 i/s - 10.53x  (± 0.00) slower
```

Reply:
If the spec says it then fine by me.

However note that your benchmark isn't benching what you think it is benching. `[key: val]` is shorthand for `[{key: val}]`, but more importantly, since you use `sort!`, you are testing the happy path.

Anyway.

Reply:
Ah yes 🤦‍♂️ thanks for the correction.